core/jvm/src/SegmentPool.kt

```diff
@@ -25,34 +25,81 @@ import kotlinx.io.SegmentPool.LOCK
 import kotlinx.io.SegmentPool.MAX_SIZE
 import kotlinx.io.SegmentPool.recycle
 import kotlinx.io.SegmentPool.take
-import java.util.concurrent.atomic.AtomicReference
+import java.util.concurrent.atomic.AtomicIntegerFieldUpdater
+import java.util.concurrent.atomic.AtomicReferenceArray
+
+/**
+ * Precise [SegmentCopyTracker] implementation tracking a number of shared segment copies.
+ * Every [addCopy] call increments the counter, every [removeCopyIfShared] decrements it.
+ *
+ * After calling [removeCopyIfShared] the same number of time [addCopy] was called, tracker returns to the unshared state.
+ *
+ * The class is internal for testing only.
+ */
+internal class RefCountingCopyTracker : SegmentCopyTracker() {
+    companion object {
+        @JvmStatic
+        private val fieldUpdater = AtomicIntegerFieldUpdater.newUpdater(RefCountingCopyTracker::class.java, "refs")
+    }
+
+    @Volatile
+    private var refs: Int = 0
```

Comment:
Please, rename property to `copyCount` or something like that, because by the name of `refs` it seems that this is the total number of segments